src/number_theory/fermat4.lean

```diff
@@ -33,18 +34,14 @@ begin
   { intro f42,
     split, { exact mul_ne_zero hk0 f42.1 },
     split, { exact mul_ne_zero hk0 f42.2.1 },
-    { calc (k * a) ^ 4 + (k * b) ^ 4
-          = k ^ 4 * (a ^ 4 + b ^ 4) : by ring
-      ... = k ^ 4 * c ^ 2 : by rw f42.2.2
-      ... = (k ^ 2 * c) ^ 2 : by ring }},
+    { have H : a ^ 4 + b ^ 4 = c ^ 2 := f42.2.2,
+      linear_combination (H, k ^ 4) } },
   { intro f42,
     split, { exact right_ne_zero_of_mul f42.1 },
     split, { exact right_ne_zero_of_mul f42.2.1 },
     apply (mul_right_inj' (pow_ne_zero 4 hk0)).mp,
-    { calc k ^ 4 * (a ^ 4 + b ^ 4)
-          = (k * a) ^ 4 + (k * b) ^ 4 : by ring
-      ... = (k ^ 2 * c) ^ 2 : by rw f42.2.2
-      ... = k ^ 4 * c ^ 2 : by ring }}
+    have H : (k * a) ^ 4 + (k * b) ^ 4 = (k ^ 2 * c) ^ 2 := f42.2.2,
+    linear_combination H }
```

Comment:
This suggests a generalization of `linear_combination`: take expressions instead of hypothesis names, so you could write `linear_combination f42.2.2`.